### PR TITLE
Added and improved examples of code in application

### PR DESCRIPTION
**Pull request by martinstarkov**
Added a bunch of examples for the different object types and fixed inconsistent formatting to match one style.

### CODE REVIEW

File: source/examples/queue-parallel.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: 2020 The Khronos Group Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <CL/sycl.hpp>
+#include <iostream>
+
+const int count = 10;
```
Comment (keryell) [inline source/examples/queue-parallel.cpp:8]:
```suggestion
constexpr int count = 10;
```

File: source/examples/queue-parallel.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                                         auto index = item.get_id(0);
+                                         data_accessor[index] += 5;
+                                       });
+    });
+    q.wait();
+  }
+
+  for (int i = 0; i < count; ++i) {
+    std::cout << data[i] << " ";
+  }
```
Comment (keryell) [inline source/examples/queue-parallel.cpp:33]: There is a lacking end-of-line.
You could also use a for-range-loop like:
```suggestion
  for (auto e : data)
    std::cout << data[i] << " ";
  std::cout << std::endl;
```

File: source/examples/queue-single-task.cpp [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+#include <CL/sycl.hpp>
+
+int main() {
+
+  auto q = sycl::queue(sycl::default_selector());
+
+  q.submit([&](sycl::handler &cgh) {
+    auto os = sycl::stream{128, 128, cgh};
+    cgh.single_task<class hello_world>([=]() { os << "Hello World!\n"; });
```
Comment (keryell) [inline source/examples/queue-single-task.cpp:13]:
```suggestion
    cgh.single_task<class hello_world>([=] { os << "Hello World!\n"; });
```
Be lazy...

**Review by keryell — COMMENTED** [review]
Interesting

File: source/examples/queue-parallel.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: 2020 The Khronos Group Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <CL/sycl.hpp>
```
Comment (keryell) [inline source/examples/queue-parallel.cpp:5]: To update to SYCL 2020...

Comment (martinstarkov) [reply source/examples/queue-parallel.cpp:5]: Not sure what you mean by this comment, could you clarify @keryell?

Comment (keryell) [reply source/examples/queue-parallel.cpp:5]: This is
```c++
#include <sycl/sycl.hpp>
```
in SYCL 2020.

Comment (martinstarkov) [reply source/examples/queue-parallel.cpp:5]: Ah, understood. I'll fix that.

**Comment by martinstarkov** [comment]
Closing this and splitting changes to smaller pull requests for easier review and keeping track of history

**Pull request closed without merging**